fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MultiDistinctGroupConcat.java

```diff
@@ -0,0 +1,164 @@
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(VarcharType.SYSTEM_DEFAULT).args(VarcharType.SYSTEM_DEFAULT),
+            FunctionSignature.ret(VarcharType.SYSTEM_DEFAULT).varArgs(VarcharType.SYSTEM_DEFAULT,
+                    AnyDataType.INSTANCE),
+            FunctionSignature.ret(VarcharType.SYSTEM_DEFAULT).varArgs(VarcharType.SYSTEM_DEFAULT,
+                    VarcharType.SYSTEM_DEFAULT, AnyDataType.INSTANCE));
+
+    private final int nonOrderArguments;
+
+    /**
+     * constructor with 1 argument.
+     */
+    public MultiDistinctGroupConcat(boolean alwaysNullable, Expression arg,
+            OrderExpression... orders) {
+        super("multi_distinct_group_concat", true, alwaysNullable,
+                ExpressionUtils.mergeArguments(arg, orders));
+        this.nonOrderArguments = 1;
+    }
+
+    /**
+     * constructor with 1 argument.
+     */
+    public MultiDistinctGroupConcat(Expression arg, OrderExpression... orders) {
+        this(false, arg, orders);
+    }
+
+    /**
+     * constructor with 2 arguments.
+     */
+    public MultiDistinctGroupConcat(boolean alwaysNullable, Expression arg0,
+            Expression arg1, OrderExpression... orders) {
+        super("multi_distinct_group_concat", true, alwaysNullable,
+                ExpressionUtils.mergeArguments(arg0, arg1, orders));
+        this.nonOrderArguments = 2;
+    }
+
+    /**
+     * constructor with 2 arguments.
+     */
+    public MultiDistinctGroupConcat(Expression arg0, Expression arg1, OrderExpression... orders) {
+        this(false, arg0, arg1, orders);
+    }
+
+    /**
+     * constructor for always nullable.
+     */
+    public MultiDistinctGroupConcat(boolean alwaysNullable, int nonOrderArguments,
+            List<Expression> args) {
+        super("multi_distinct_group_concat", true, alwaysNullable, args);
+        this.nonOrderArguments = nonOrderArguments;
+    }
+
+    @Override
+    public boolean nullable() {
+        return children().stream()
+                .anyMatch(expression -> !(expression instanceof OrderExpression) && expression.nullable());
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        DataType typeOrArg0 = getArgumentType(0);
+        if (!typeOrArg0.isStringLikeType() && !typeOrArg0.isNullType()) {
+            throw new AnalysisException(
+                    "multi_distinct_group_concat requires first parameter to be of type STRING: "
+                            + this.toSql());
+        }
+
+        if (nonOrderArguments == 2) {
+            DataType typeOrArg1 = getArgumentType(1);
+            if (!typeOrArg1.isStringLikeType() && !typeOrArg1.isNullType()) {
+                throw new AnalysisException(
+                        "multi_distinct_group_concat requires second parameter to be of type STRING: "
+                                + this.toSql());
+            }
+        }
+    }
```

Comment:
i don't think it could work as u expect since `checkLegalityBeforeTypeCoercion` only called in `FunctionBinder`